mithril-client/src/cardano_database_client/proving.rs

```diff
@@ -106,32 +107,39 @@ impl InternalArtifactProver {
                 DigestLocation::CloudStorage { .. } | DigestLocation::Aggregator { .. } => {
                     self.http_file_downloader.clone()
                 }
-            };
-            let file_downloader_uri: FileDownloaderUri = location.into();
-            let downloaded = file_downloader
-                .download_unpack(
-                    &file_downloader_uri,
-                    digest_file_target_dir,
-                    None,
-                    DownloadEvent::Digest {
-                        download_id: download_id.clone(),
-                    },
-                )
-                .await;
-            match downloaded {
-                Ok(_) => {
-                    self.feedback_sender
-                        .send_event(MithrilEvent::CardanoDatabase(
-                            MithrilEventCardanoDatabase::DigestDownloadCompleted { download_id },
-                        ))
-                        .await;
-                    return Ok(());
+                DigestLocation::Unknown => {
+                    return Err(anyhow!("Unknown location type to download digest"));
                 }
-                Err(e) => {
-                    slog::error!(
-                        self.logger,
-                        "Failed downloading and unpacking digest for location {file_downloader_uri:?}"; "error" => ?e
-                    );
+            };
+            let file_downloader_uri: StdResult<FileDownloaderUri> = location.try_into();
+            if let Ok(file_downloader_uri) = file_downloader_uri {
```

Comment:
You could handle the same way as in `download_unpack`:
```suggestion
            let file_downloader_uri: StdResult<FileDownloaderUri> = location.try_into()?;
```

Reply:
Done